### DISCUSSION TITLE
Interpreting ANCOMBC2 results for a continuous variable

### DISCUSSION THREAD

**Discussion by LLansing** [FrederickHuangLin/ANCOMBC#222] [General]
Hello,

In the experiment I'm working with, I wish to see which taxa are correlated with the factor of time. I have 8 timepoints from which my samples are taken, and I do not want to treat them as a categorical variable, since I am interested in taxa that are differentially abundant across time rather than between particular timepoints.

**I've ran _ancombc2_ (and tested with _ancombc_ as well) while treating the time variable as continuous, but I've been unable to find in the ancombc documentation or paper how to interpret the results (log-fold change, q_val) for a continuous variable. Could you please help me understand?**

---
Additionally, I ask for your opinion on the following. The lfc values of the significant diff taxa are much lower than I would have expected, compared to the other non-_time_ factors which were included in the formula. You can see in the following volcano plot that there are many taxa with strong q_values but weak lfc.
![image](https://github.com/FrederickHuangLin/ANCOMBC/assets/35748909/32ae278f-09f2-4013-9325-f1830d4395ac)

Plotting the raw reads over time of the significantly different taxa reveals that a number of them are taxa which have non-zero counts at early timepoints but have 0 (or very low) counts at later timepoints. (note: in the image, only one dot appears for later timepoints on the x-axis, but this is several samples overlapping)
![image](https://github.com/FrederickHuangLin/ANCOMBC/assets/35748909/231630b2-0325-4e11-a751-e6061b04bb6f)
I would have expected a stronger lfc for such taxa, given the strong drop off in counts over time. Am I misguided in this assumption?

**Comment by FrederickHuangLin** [comment]
Hello @LLansing,

Interpreting log fold-change and q-values for continuous variables is quite similar to how it's done for categorical variables. However, for continuous variables, the effect sizes correspond to a "one unit increase" in the continuous variable.

The difference between the log fold-change and the results you observed using the raw data likely arises from the fact that ANCOM-BC2 operates on bias-corrected abundances and, by default, works on complete data (excluding zeros). The analysis of zero counts is evaluated in pseudo-count sensitivity analysis.

If you have any further questions, please feel free to reach out.

Best regards,
Huang

**Reply by LLansing** [reply]
Hi @FrederickHuangLin, 
Thank you for your response. I understand better now. The effect-size per "one-unit-increase" would seem to explain the small values in my volcano plot x-axis, as the effect per unit is quite small.
I've also discovered how to retrieve the the bias-corrected abundances from the ANCOMBC2 results, which I can use for plots to visualize the change in time of differentially abundant taxa while corresponding more directly to the ANCOMBC2 test result.

**Reply by LLansing** [reply]
@FrederickHuangLin 
I have a follow-up about interpretation of ANCOMBC2 with continuous variables.
Am I able to extrapolate the effect-size value to more units of the continuous variable? For example, if I want to find the log fold-change for 10 units, would I simply multiply the per-unit log-fold value by 10?

**Reply by FrederickHuangLin** [reply]
You are absolutely right, @LLansing :)

**Comment by sophia-richter** [comment]
Hi @FrederickHuangLin I am still a bit confused about how to interpret log change when using a continuous variable. Does the log fold change mean that the taxa changed that "amount" between the lowest and highest values along the continuous range? So roughly a taxa with positive log fold change increases as the continuous variable increases, and a taxa with negative log fold change decreases as the continuous variable decreases?

I guess I am just a bit confused on how ANCOM runs on continuous variables in general - is it looking only at linear trends of taxon abundance?

Additionally, I noticed that many of my taxa that are differentially abundant do not pass the pseudo-count sensitivity test. How important is this in terms of reporting / considering results significant or not?

Thank you so much!
